### ISSUE TITLE
Hardware: Arduino, ESP8266, BeagleBone

### ISSUE THREAD

**Issue by ches** [bkkhack/hacknights#8]
I have some hardware toys that I normally bring every month. These include an [Arduino Mega 2560](https://www.arduino.cc/en/Main/ArduinoBoardMega2560) and a [BeagleBone Black](http://beagleboard.org/BLACK), plus some basic components for doing introductory projects (sensors and actuators, breadboards and plenty of wires).

I may or may not do hardware projects at hack night, but there are usually a few people who do; you are welcome to use my gear, and use this ticket to discuss teaming up together.

There are endless free project tutorials on the web and I have a few intro guide books I can share with you as PDFs too. Some great places to start if you're beginning:
- [Arduino Starter Kit video tutorials by Arduino creator Massimo Banzi](https://www.youtube.com/playlist?list=PLT6rF_I5kknPf2qlVFlvH47qHvqvzkknd)
- [Arduino Wiki tutorials](https://www.arduino.cc/en/Tutorial/HomePage)
- BeagleBone has a bunch of docs shipped onboard, see the [Getting Started](http://beagleboard.org/getting-started) page
- Adafruit's learning section, such as [the BeagleBone category](https://learn.adafruit.com/category/beaglebone)

**Comment by scorpiogit** [comment]
Looks good, am interested, have some kit at home - but probably heading home early tonight.

**Comment by pairochjulrat** [comment]
That's  interested , I started play with Arduino and Esp8266  a few month.
 That things is very cool tiny stuff . So hope to join and share next time.

**Comment by ches** [comment]
@scorpiogit: No worries, like I said I try to bring the stuff every month and there are usually a few regulars playing with it.

@pairochjulrat Nice, I've read about the ESPs a bit and wish I had one to bring, seems like a really cool platform with big specs for small boards, and wifi built-in. Bring it in any time!

**Comment by bighappyworld** [comment]
I've got a raspberry pi and trying to keep it connected to a websocket on a remote server but it disconnects about every 24+ hours. Any thoughts?

**Comment by salinceva** [comment]
In

**Comment by ches** [comment]
I got a new toy for the collection, a [NodeMCU DevKit](http://www.nodemcu.com/index_en.html) which is a little board based on the ESP8266 chip that became all the rage over the past year or so. Built-in wifi—in fact it can be a client and an access point at the same time—more power and program storage than the basic Arduinos, enough I/O to do a lot of projects, and you can get them for a few dollars. It's pretty fun.

The NodeMCU project has their own firmware where you program in Lua (you can run an interactive Lua interpreter straight off the board over serial, so a more interactive development experience is possible for experimenting than what's typical with microcontrollers), or there is support for developing with the Arduino IDE using most Arduino libraries. There's even [MicroPython](http://docs.micropython.org/en/latest/esp8266/) as an option, which I haven't had the chance to try yet.

I've ordered a [WeMos D1 Mini](http://www.wemos.cc/Products/d1_mini.html) also, another ESP8266 board, with some of its shields (interested in SD card and battery power capability for a data logger), but it's coming from AliExpress so it won't be here for awhile.

I ordered a few parts recently from [Arduitronics](http://arduitronics.com/) in Bangkok and I can recommend them if they have what you're looking for. Processed my order first thing in the morning and had it delivered by Kerry Logistics courier the same afternoon. Got my NodeMCU board from them for about the same as it would have cost from Ali, they have the base WeMos board too but none of the shields for it.

**Comment by pairochjulrat** [comment]
Yes the  NodeMCU is quite popular of esp8266 breakout board but also there are a lot of fake NodeMCU out there, so  please be aware :P

This link may help to notice when get that thing: http://frightanic.com/iot/comparison-of-esp8266-nodemcu-development-boards/

I have a little blog: http://roiet.io for memorise the hobby with my son, of course including the esp8266 projects. Hope to meet up with you guy about those devices someday.

**Comment by ches** [comment]
> This link may help to notice when get that thing: http://frightanic.com/iot/comparison-of-esp8266-nodemcu-development-boards/

Yeah I read this guide, it is very useful, thanks for sharing here. Technically it is open-source hardware so I find it a bit odd that the NodeMCU people have sometimes referred to "fakes" ("low-quality copies" might be more apt, maybe a native language subtlety). I would definitely suggest that people get a "v2" / DevKit 1.0 board though rather than the "v3" ones on Ali that are third-party and lower-cost—even if the quality of the "v3" ones is fine, they are larger and completely cover the width of a breadboard like the first NodeMCU DevKit 0.9 did. The v2 leaves a row of pins on either side which is way, way more convenient. Arduitronics stocks the v2 locally.

> I have a little blog: http://roiet.io for memorise the hobby with my son, of course including the esp8266 projects. Hope to meet up with you guy about those devices someday.

Very cool! Hope you can join the hack nights sometime also.

**Comment by Alkia** [comment]
I'm a big fan of NodeMCU using arduino IDE rather than Lua for it. Willing to share on it.

**Comment by bhargavg** [comment]
Hey, I would like to join, is this group still active?

**Comment by salinceva** [comment]
Hi bhargavg, yes.. lets start our own hardware gang..

Ive been wanting to setup the esp8266 and arduino with some servos for a quadcopter so ill bring that stuff. Feel free to bring anything else ..

see you on Thursday.

**Comment by bhargavg** [comment]
@salinceva Unfortunately I don't have any kits with me as of now. Could you please suggest what will be the best way to procure electronic kits in Bangkok?

**Comment by bmason** [comment]
I am interested in a reliable source as well.

**Comment by salinceva** [comment]
@bhargavg @bmason if you guys dont care about speed i usually just order via ebay takes ~ 14 days
if you want things sooner there is  www.thaieasyelec.com/  or  www.homeofmaker.com/
i think @ches usually brings some gear and ill bring whatever i have also so we should be fine for messing around on some projects..

**Comment by bhargavg** [comment]
I got my ESP32 board from ebay last week. I'm planning to make it into an iBeacon and use for  indoor navigation to track my position in home and hopefully turn on/off lights based on that.

**Comment by salinceva** [comment]
sounds great.. im going to try to recreate the light sensor that you 
made last month.


Bhargav Gurlanka wrote:

**Comment by ches** [comment]
@bhargavg @salinceva I left the house today without my bag of gear, going home for it isn't really feasible so I hope you're able to bring yours 😅
